diff --git a/gsi_car_base.mk b/gsi_car_base.mk
old mode 100644
new mode 100755
index 85d1277..8f3af6f
--- a/gsi_car_base.mk
+++ b/gsi_car_base.mk
@@ -57,6 +57,7 @@ PRODUCT_ARTIFACT_PATH_REQUIREMENT_ALLOWED_LIST += \
     system/etc/permissions/com.android.car.activityresolver.xml \
     system/etc/permissions/com.android.car.bugreport.xml \
     system/etc/permissions/com.android.car.carlauncher.xml \
+    system/etc/permissions/com.android.car.mccarlauncher.xml \
     system/etc/permissions/com.android.car.cluster.home.xml \
     system/etc/permissions/com.android.car.dialer.xml \
     system/etc/permissions/com.android.car.hvac.xml \
@@ -105,6 +106,7 @@ PRODUCT_ARTIFACT_PATH_REQUIREMENT_ALLOWED_LIST += \
     system/priv-app/CarDialerApp/CarDialerApp.apk \
     system/priv-app/CarHvacApp/CarHvacApp.apk \
     system/priv-app/CarLauncher/CarLauncher.apk \
+    system/priv-app/CarLauncher/MCCarLauncher.apk \
     system/priv-app/CarMediaApp/CarMediaApp.apk \
     system/priv-app/CarMessengerApp/CarMessengerApp.apk \
     system/priv-app/CarRadioApp/CarRadioApp.apk \
